### PR TITLE
Update misc configuration

### PR DESCRIPTION
**Pull request by li-kai**
1. Remove travis config
1. Rename /docs to /misc.
    I've repeatedly thought to myself "Huh, since when did we have docs?"
    This makes the intent of the folder more explicit
1. Removes the duplicated nvmrc, telling netlify to use dubnium
1. Updates renovate config to group monorepos and etc together so we have less of a burden
    This is the reason behind the PR title.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
     "config:base",
     "schedule:earlyMondays",
     "group:linters",
+    "group:monorepos",
+    "group:definitelyTyped",
```
Comment (ZhangYiJiang) [inline renovate.json]: I don’t think this is a good idea. We want to group the type with their dependents, and not upgrade all types all at once

File: renovate.json [hunk 2]
```diff
@@ -11,21 +12,61 @@
...
+    {
+      "description": "Group stylelint packages together",
+      "packagePatterns": ["stylelint"],
+      "excludePackageNames": ["stylelint-webpack-plugin"],
+      "groupName": "stylelint"
+    },
+    {
+      "description": "Group test-related packages together",
+      "extends": "monorepo:jest",
+      "packagePatterns": ["(jest|enzyme|mock|nightwatch)"],
```
Comment (ZhangYiJiang) [inline renovate.json]: Nightwatch should probably not be bundled here.

File: renovate.json [hunk 3]
```diff
@@ -11,21 +12,61 @@
...
+      "packageNames": ["identity-obj-proxy"],
+      "groupName": "test"
+    },
+    {
+      "description": "Group test-related packages together",
```
Comment (ZhangYiJiang) [inline renovate.json]: Copy-paste error? Also if we're extending can we just leave out the description?

File: renovate.json [hunk 4]
```diff
@@ -11,21 +12,61 @@
...
+      "extends": "monorepo:react",
+      "packageNames": ["@types/react", "@types/react-dom"],
+      "groupName": "react"
+    },
+    {
+      "description": "Group test-related packages together",
```
Comment (ZhangYiJiang) [inline renovate.json]: Also copy paste error?

File: renovate.json [hunk 5]
```diff
@@ -11,21 +12,61 @@
...
+      "groupName": "react-router"
+    },
+    {
+      "description": "Group state management packages together",
+      "packagePatterns": ["(redux|reselect|immer)"],
+      "excludePackagePatterns": ["redux-mock"],
+      "groupName": "state-management"
     }
   ],
-  "ignoreDeps": [
```
Comment (ZhangYiJiang) [inline renovate.json]: Can we ignore `query-string`, since we can't upgrade past 5.x

File: renovate.json (hunk 2 above)
Comment (ZhangYiJiang) [reply renovate.json]: As an aside, 1.x was just released and comes with a new API, so upgrading to that will be a bit troublesome

Comment (li-kai) [reply renovate.json]: Fixed in #1563

File: renovate.json (hunk 5 above)
Comment (li-kai) [reply renovate.json]: Ignored in #1563

**Pull request merged**